src/recipe.data.ts

```diff
@@ -37,15 +40,35 @@ export function findAllRecipes(): Promise<Array<Recipe> | null> {
   return recipeModel.find().lean().exec();
 }
 
-export function searchForRecipe(ingred: string): Promise<Array<Recipe> | null> {
+export function searchForRecipeIngredient(
+  ingred: string
+): Promise<Array<Recipe> | null> {
```

Comment:
This should not return null.  If there are no results, the return value should just be an empty array